ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/AttestationUtil.java

```diff
@@ -120,6 +120,7 @@ public static Bitlist getAggregationBits(int committeeSize, int indexIntoCommitt
 
   public static AttestationData completeAttestationCrosslinkData(
       BeaconState state, AttestationData attestation_data, Committee committee) {
+    attestation_data.setIndex(committee.getIndex());
```

Comment:
I think we already do this here. https://github.com/PegaSysEng/artemis/pull/999/files#diff-769e7d53286f8bf0620aceb4ec1632d9R234

Reply:
ah, I thought we'd set it.  The reason that line doesn't work is it's setting it on the generic attestation data before it's copied.  But because we're using multiple threads to create attestations for each validator, there's a race condition where each thread is trying to set the index for that validator and another thread may change it again before it actually gets copied.

Setting the index in `completeAttestationCrosslinkData` works on a copy of the data specific to this validator so avoids the race condition.

Reply:
Does this fix the "Verify aggregate signature" attestation error?

Reply:
I would also move the setting of index to again the produceAttestation function, but after the creation of the new AttestationData object, since we don't really have any crosslink information, and thats why having a seperate function for that doesn't make too much sense.

Reply:
Sorry Cem, didn't see these comments.  Yes this fixes the verify aggregate signature error.  Will raise a separate PR to remove all the remaining Crosslink references and will inline the call in that.  Tempted to do it after merging back to master though to minimise conflicts.

Reply:
Agreed 👍